pkg/sql/opt/exec/execbuilder/tests/local/local_test: TestExecBuild_explain_analyze failed

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_explain_analyze [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8856279?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8856279?buildTab=artifacts#/) on master @ [afb68209901012b4147febc6d82138774b23bc3f](https://github.com/cockroachdb/cockroach/commits/afb68209901012b4147febc6d82138774b23bc3f):


```
=== RUN   TestExecBuild_explain_analyze
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/ec6a1341baa02e55a69448c66fcc4415/logTestExecBuild_explain_analyze3314418279
    test_log_scope.go:79: use -show-logs to present logs inline
    logic.go:2932: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5957/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/explain_analyze:127: EXPLAIN (VERBOSE) SELECT * FROM system.privileges WHERE path = 'vtable/crdb_internal/tables';
        expected:
            distribution: local
            vectorized: true
            ·
            • scan
              columns: (username, path, privileges, grant_options, user_id)
              estimated row count: 10 (missing stats)
              table: privileges@privileges_path_user_id_key
              spans: /"vtable/crdb_internal/tables"-/"vtable/crdb_internal/tables"/PrefixEnd
        but found (query options: "") :
            distribution: local
            vectorized: true
            ·
            • scan
              columns: (username, path, privileges, grant_options, user_id)
              estimated row count: 1 (100% of the table; stats collected <hidden> ago)
              table: privileges@privileges_path_user_id_key
              spans: /"vtable/crdb_internal/tables"-/"vtable/crdb_internal/tables"/PrefixEnd
[10:08:16] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5957/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/explain_analyze with config local: 8 tests, 1 failures
[10:08:18] --- total progress: 8 statements
--- total: 8 tests, 1 failures
    logic.go:4199: -- test log scope end --
--- FAIL: TestExecBuild_explain_analyze (7.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_explain_analyze.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-24835

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -1676,6 +1676,11 @@ func (t *logicTest) newCluster(
 		); err != nil {
 			t.Fatal(err)
 		}
+		if _, err := conn.Exec(
+			"SET CLUSTER SETTING sql.stats.system_tables_autostats.enabled = false",
+		); err != nil {
+			t.Fatal(err)
+		}
 
 		// We also disable stats forecasts to have deterministic tests. See #97003
 		// for details.

